pkg/kubectl/scale.go

```diff
@@ -184,14 +201,87 @@ type realScalerClient struct {
 	client client.Interface
 }
 
+// GetReplicationController returns the replication controller with the provided namespace/name
 func (c *realScalerClient) GetReplicationController(namespace, name string) (*api.ReplicationController, error) {
 	return c.client.ReplicationControllers(namespace).Get(name)
 }
 
+// UpdateReplicationController updates the provided replication controller in the given namespace
 func (c *realScalerClient) UpdateReplicationController(namespace string, rc *api.ReplicationController) (*api.ReplicationController, error) {
 	return c.client.ReplicationControllers(namespace).Update(rc)
 }
 
-func (c *realScalerClient) ControllerHasDesiredReplicas(rc *api.ReplicationController) wait.ConditionFunc {
-	return client.ControllerHasDesiredReplicas(c.client, rc)
+// WaitForReplicas is a closure around the WaitForReplicas function
+func (c *realScalerClient) WaitForReplicas(rc *api.ReplicationController, delta int, timeout time.Duration) error {
+	return WaitForReplicas(c.client, rc, delta, timeout)
+}
+
+// WaitForReplicas accepts a replication controller and waits until it either observes delta pod events
+// (additions, deletions) or times out
+func WaitForReplicas(client client.Interface, rc *api.ReplicationController, delta int, timeout time.Duration) error {
+	done := make(chan struct{})
+	stopCh := make(chan struct{})
+	changes := 0
+
+	switch {
+	case delta < 0:
+		delta *= -1
+	case delta == 0:
+		return nil
+	}
+
+	checkPods := func(obj interface{}) {
+		changes++
+		if changes == delta {
+			safeClose(stopCh)
+			done <- struct{}{}
+		}
+	}
+
+	_, controller := framework.NewInformer(
+		createPodLWForRC(client, rc),
+		&api.Pod{},
+		time.Second*30,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    checkPods,
+			DeleteFunc: checkPods,
+		},
+	)
+
+	go controller.Run(stopCh)
```

Comment:
so we still have a race, please add a // TODO: checking for 0 pods isn't sufficient here, because the rc could still create pods between this check and when we delete  it. Fix this with graceful delete #9147